self-hosted-login/okta-aspnetcore-mvc-example/Controllers/AccountController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Mvc;
+
+namespace okta_aspnetcore_mvc_example.Controllers
+{
+    public class AccountController : Controller
+    {
+        public IActionResult Login()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Login([FromForm]string sessionToken)
+        {
+            if (!HttpContext.User.Identity.IsAuthenticated)
+            {
+                var properties = new AuthenticationProperties();
+                properties.Items.Add("sessionToken", sessionToken);
+                properties.RedirectUri = "/Home/About";
+
+                return Challenge(properties, OpenIdConnectDefaults.AuthenticationScheme);
+            }
+
+            return RedirectToAction("Index", "Home");
+        }
+
+        [HttpPost]
+        public IActionResult Logout()
+        {
+            return new SignOutResult(new[] { OpenIdConnectDefaults.AuthenticationScheme, CookieAuthenticationDefaults.AuthenticationScheme });
```

Comment:
Use `OktaDefaults.MvcAuthenticationScheme`

Can you check if order matters here? I think it's better to list `CookieAuthenticationDefaults` first, because then it signs out of the cookie scheme first. I'm not 100% sure though.